lib/eventFetch.js

```diff
@@ -31,32 +31,35 @@ export function mapLinkList(linkList) {
   return [];
 }
 
-export function mapAndSanitateEvent(event) {
+export function mapAndSanitate(item, type) {
+  const imageField = (type === 'news') ? 'featureImage' : 'event-image'
+
   return {
-    id: event.id,
-    slug: event.slug ? event.slug : null,
-    tags: event.tags || [],
-    title: event.data['event.title'] ?
-      event.data['event.title'].value : null,
-    strapline: event.data['event.strapline'] ?
-      event.data['event.strapline'].value : null,
-    datetime: event.data['event.timestamp'] ?
-      mapEventDate(event.data['event.timestamp'].value) : null,
-    internalLinks: event.data['event.internalLinks'] ?
-      mapLinkList(event.data['event.internalLinks'].value) : [],
-    externalLinks: event.data['event.externalLinks'] ?
-      mapLinkList(event.data['event.externalLinks'].value) : [],
-    body: event.data['event.body'] ?
-      event.data['event.body'].value : [],
-    featureImageFilename: event.data['event.event-image'] ?
-      event.data['event.event-image'].value : null,
+    id: item.id,
+    slug: item.slug ? item.slug : null,
+    tags: item.tags || [],
+    title: item.data[type + '.title'] ?
+      item.data[type + '.title'].value : null,
+    strapline: item.data[type + '.strapline'] ?
+      item.data[type + '.strapline'].value : null,
+    datetime: item.data[type + '.timestamp'] ?
+      mapEventDate(item.data[type + '.timestamp'].value) : null,
+    internalLinks: item.data[type + '.internalLinks'] ?
+      mapLinkList(item.data[type + '.internalLinks'].value) : [],
+    externalLinks: item.data[type + '.externalLinks'] ?
+      mapLinkList(item.data[type + '.externalLinks'].value) : [],
+    body: item.data[type + '.body'] ?
+      item.data[type + '.body'].value : [],
+    featureImageFilename: item.data[type + '.' + imageField] ?
+      item.data[type + '.' + imageField].value : null,
   };
 }
 
 export async function EventFetch(_, args, req) {
   const ref = req.headers['x-preview'];
+  const docType = (this && this.docType) ? this.docType : 'event';
 
-  const r = await Prismic.api(PrismicConfig.apiEndpoint).then((api) =>
+  const res = await Prismic.api(PrismicConfig.apiEndpoint).then((api) =>
```

Comment:
We should be using the `async`/`await` *or* Promises as a pattern, and not both. Feels it introduces an unneccessary level of confusion, otherwise